src/ZeroC.Slice/SliceDecoderExtensions.cs

```diff
@@ -123,7 +123,7 @@ public static T[] DecodeSequence<T>(this ref SliceDecoder decoder, Action<T>? ch
             int elementSize = Unsafe.SizeOf<T>();
             decoder.IncreaseCollectionAllocation(count * elementSize);
             var value = new T[count];
-            Span<byte> destination = MemoryMarshal.Cast<T, byte>(value);
+            Span<byte> destination = MemoryMarshal.Cast<T, byte>(value.AsSpan());
```

Comment:
Without this AsSpan, we end up with another Cast overload that returns a ReadOnlySpan.